lib/core/test/unit/Cardano/Wallet/DB/SqliteFileModeSpec.hs

```diff
@@ -0,0 +1,284 @@
+                (`readPrivateKey` testPk)
+                (Just (k, h))
+                Nothing )
+
+        it "put and read tx history" $ \(conn,db) -> do
+            unsafeRunExceptT $ createWallet db testPk testCp testMetadata
+            runExceptT (putTxHistory db testPk testTxs) `shouldReturn` Right ()
+            readTxHistory db testPk `shouldReturn` testTxs
+            close conn
+            ( testOpeningCleaning
+                (`readTxHistory` testPk)
+                testTxs
+                Map.empty )
+
+        it "put and read checkpoint" $ \(conn,db) -> do
+            unsafeRunExceptT $ createWallet db testPk testCp testMetadata
+            runExceptT (putCheckpoint db testPk testCp) `shouldReturn` Right ()
+            readCheckpoint db testPk `shouldReturn` Just testCp
+            close conn
+            ( testOpeningCleaning
+                (`readCheckpoint` testPk)
+                (Just testCp)
+                Nothing )
+
+    withDB inMemoryDBLayer $
+        describe "random operation chunks property when writing to/reading from file" $ do
+        it "realize a random batch of operations upon one db open"
+            (property . prop_randomOpChunks)
+
+    where
+        testOpeningCleaning
+            :: (Show s, Eq s)
+            => (DBLayer IO (SeqState DummyTarget) DummyTarget -> IO s)
+            -> s
+            -> s
+            -> Expectation
+        testOpeningCleaning call expectedAfterOpen expectedAfterClean = do
+            (conn1, db1) <- fileDBLayer
+            call db1 `shouldReturn` expectedAfterOpen
+            _ <- cleanDB db1
+            call db1 `shouldReturn` expectedAfterClean
+            close conn1
+            (conn2,db2) <- fileDBLayer
+            call db2 `shouldReturn` expectedAfterClean
+            close conn2
+
+        openCloseDB :: IO ()
+        openCloseDB = do
+            (conn, _) <- fileDBLayer
+            close conn
+
+        dbCleaned
+            :: (Connection, DBLayer IO (SeqState DummyTarget) DummyTarget)
+            -> IO  (Connection, DBLayer IO (SeqState DummyTarget) DummyTarget)
+        dbCleaned (conn, db) = do
+            _ <- cleanDB db
+            pure (conn, db)
+
+
+-- This property checks that executing series of wallet operations in a single
+-- SQLite session has the same effect as executing the same operations over
+-- multiple sessions.
+prop_randomOpChunks
+    :: DBLayer IO (SeqState DummyTarget) DummyTarget
+    -> KeyValPairs (PrimaryKey WalletId) (Wallet (SeqState DummyTarget) DummyTarget , WalletMetadata)
+    -> Property
+prop_randomOpChunks inMemoryDB (KeyValPairs pairs) =
+    not (null pairs) ==> monadicIO (pure inMemoryDB >>= prop)
+  where
+    prop dbM = liftIO $ do
+        (conn, dbF) <- fileDBLayer
+        _ <- cleanDB dbF
+        _ <- cleanDB inMemoryDB
+
+        forM_ pairs (updateDB dbM)
+        chunks <- cutRandomly [] pairs
+        forM_ chunks handleChunks
+
+        verify dbM dbF
+
+        close conn
```

Comment:
comment this and add `withMaxSuccess 1000` and will probably end up with : 
```
     uncaught exception: SqliteException
       SQLite3 returned ErrorCan'tOpen while attempting to perform open "backup/test.db".
       (after 504 tests and 4 shrinks)
```
as I did in my laptop.

It suggests we need closing when using file-based Sqlite